vllm/entrypoints/chat_utils.py

```diff
@@ -187,15 +146,42 @@ def _parse_chat_message_content(
 ) -> ChatMessageParseResult:
     role = message["role"]
     content = message.get("content")
+    tool_call_id = message.get('content')
+    tool_calls = message.get('tool_calls')
+    # no longer used by OpenAI, but some models still use it for tool calls.
+    name = message.get('name', '')
 
-    if content is None:
+    # empty case
+    if content is None and tool_calls is None:
         return ChatMessageParseResult(messages=[], mm_futures=[])
-    if isinstance(content, str):
+
+    # special case - assistant message where tool calls are provided.
+    if role == 'assistant' and tool_calls is not None:
+        messages = [
+            ConversationMessage(role=role,
+                                content=cast(Optional[str], content),
```

Comment:
This is required because the type of `ChatCompletionMessageParam.content` is a Union of like 8 different types from OpenAI
![Screenshot 2024-08-07 at 4 14 46 PM](https://github.com/user-attachments/assets/217dbc87-f46f-4066-9f08-5fdb2c12b9fe).

One of these types (`ChatCompletionUserMessageParam`) allows the `ChatCompletionMessage`'s `content` field to be an array of the prompt AND the image information (`ChatCompletionContentPartParam`)  when the user passed an image in.

Here I am artificially narrowing the types to be what they are allowed to be given that the message is known to be an assistant message based on the `if` block, and therefore cannot be a `ChatCompletionUserMessageParam` and therefore `content` cannot be a `ChatCompletionContentPartParam`.